File: 93-leetcode/Arrays_Hashing/217_Contains_Duplicate.js
/**
 * @param {number[]} nums
 * @return {boolean}
 */
var containsDuplicate = function(nums) {
    let seen = {}
    let duplicate = false

    for(let i = 0; i < nums.length ; i++){
        if(seen[nums[i]]){
            duplicate = true
            break
        }
        else{
            seen[nums[i]] = true
        }
    }
    return duplicate
};

console.log(containsDuplicate([1,2,3,1]))
console.log(containsDuplicate([1,2,3,4]))
console.log(containsDuplicate([1,1,1,3,3,4,3,2,4,2]))

File: 04-neetcode250/01-Arrays and hashing/217Contains Duplicate.py
from typing import List

class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        seen = set()
        for num in nums:
            if num in seen:
                return True
            seen.add(num)
        return False

sol = Solution()

nums = [1,2,3,1]
print(sol.containsDuplicate(nums))

nums = [1,2,3,4]
print(sol.containsDuplicate(nums))

nums = [1,1,1,3,3,4,3,2,4,2]
print(sol.containsDuplicate(nums))
        